lib/yard-sorbet/sig_to_yard.rb

```diff
@@ -95,4 +92,10 @@ def self.convert_type(node)
       [node.source]
     end
   end
+
+  def self.fix_generic_aref(node)
+    return node.source if node.children.empty? || node.type != :aref
+
+    "#{node.children.first.source}[#{fix_generic_aref(node.children.last.children.first)}]"
```

Comment:
```suggestion
    collection_type = node.children.first.source
    member_type = build_generic_type(node.children.last.children.first)
    "#{collection_type}[#{member_type}]"
```
will still be more readable.